<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 2.4.0">
  <meta name="generator" content="Hugo 0.46" />
  <meta name="author" content="Erica S. Nielsen">

  
  
  
  
    
  
  <meta name="description" content="As Pooled sequencing gains popularity in the world of genomics, more and more programs are being introduced to specifically handle Pool-seq data. However, for students such as myself, it can be difficult to find the right program for each step of the bioinformatics process, and to know the correct data format and parameters used with each individual program.
Here is a short recap of the Pool-seq bioinformatic programs that I have gathered (in no particular order), including input and output files, and any additional notes I had on the program.">

  
  <link rel="alternate" hreflang="en-us" href="https://esnielsen.github.io/post/a-guide-to-ngs-pool-seq-programs/">

  


  

  
  
  
  <meta name="theme-color" content="#0095eb">
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" crossorigin="anonymous">
        
      
    

    

    

  

  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700%7cRoboto:400,400italic,700%7cRoboto&#43;Mono">
  

  <link rel="stylesheet" href="/styles.css">
  

  
  
  

  
  <link rel="alternate" href="https://esnielsen.github.io/index.xml" type="application/rss+xml" title="">
  <link rel="feed" href="https://esnielsen.github.io/index.xml" type="application/rss+xml" title="">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="https://esnielsen.github.io/post/a-guide-to-ngs-pool-seq-programs/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@https://twitter.com/erica_at_work">
  <meta property="twitter:creator" content="@https://twitter.com/erica_at_work">
  
  <meta property="og:site_name" content="">
  <meta property="og:url" content="https://esnielsen.github.io/post/a-guide-to-ngs-pool-seq-programs/">
  <meta property="og:title" content="Pool-seq Programs for Population Genomics | ">
  <meta property="og:description" content="As Pooled sequencing gains popularity in the world of genomics, more and more programs are being introduced to specifically handle Pool-seq data. However, for students such as myself, it can be difficult to find the right program for each step of the bioinformatics process, and to know the correct data format and parameters used with each individual program.
Here is a short recap of the Pool-seq bioinformatic programs that I have gathered (in no particular order), including input and output files, and any additional notes I had on the program."><meta property="og:image" content="https://esnielsen.github.io/img/headers/pic.pool.png">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2018-09-07T00:00:00&#43;00:00">
  
  <meta property="article:modified_time" content="2018-09-07T00:00:00&#43;00:00">
  

  

  

  <title>Pool-seq Programs for Population Genomics | </title>

</head>
<body id="top" data-spy="scroll" data-target="#toc" data-offset="71" >

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
      <a class="navbar-brand" href="/"></a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      
      <ul class="nav navbar-nav navbar-right">
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Tutorials</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Blog</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/files/ESN.CV.2020.pdf">
            
            <span>CV</span>
            
          </a>
        </li>

        
        

        
        
        
        
        

        <li class="nav-item">
          <a href="/">
            
            <span></span>
            
          </a>
        </li>

        
        
      

      
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  
<div class="article-header">
  
  
    <img src="/img/headers/pic.pool.png" class="article-banner" itemprop="image">
  

  
</div>



  <div class="article-container">
    <h1 itemprop="name">Pool-seq Programs for Population Genomics</h1>

    

<div class="article-metadata">

  
  
  <span itemscope itemprop="author" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="Erica S. Nielsen">
  </span>
  

  <span class="article-date">
    
    <meta content="2018-09-07 00:00:00 &#43;0000 UTC" itemprop="datePublished">
    <time datetime="2018-09-07 00:00:00 &#43;0000 UTC" itemprop="dateModified">
      Sep 7, 2018
    </time>
  </span>
  <span itemscope itemprop="publisher" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="Erica S. Nielsen">
  </span>

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    4 min read
  </span>
  

  
  
  <span class="middot-divider"></span>
  <a href="https://esnielsen.github.io/post/a-guide-to-ngs-pool-seq-programs/#disqus_thread"></a>
  

  
  
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=Pool-seq%20Programs%20for%20Population%20Genomics&amp;url=https%3a%2f%2fesnielsen.github.io%2fpost%2fa-guide-to-ngs-pool-seq-programs%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fesnielsen.github.io%2fpost%2fa-guide-to-ngs-pool-seq-programs%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=https%3a%2f%2fesnielsen.github.io%2fpost%2fa-guide-to-ngs-pool-seq-programs%2f&amp;title=Pool-seq%20Programs%20for%20Population%20Genomics"
         target="_blank" rel="noopener">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=https%3a%2f%2fesnielsen.github.io%2fpost%2fa-guide-to-ngs-pool-seq-programs%2f&amp;title=Pool-seq%20Programs%20for%20Population%20Genomics"
         target="_blank" rel="noopener">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=Pool-seq%20Programs%20for%20Population%20Genomics&amp;body=https%3a%2f%2fesnielsen.github.io%2fpost%2fa-guide-to-ngs-pool-seq-programs%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>


    <div class="article-style" itemprop="articleBody">
      <p>As Pooled sequencing gains popularity in the world of genomics, more and more programs are being introduced to specifically handle Pool-seq data. However, for students such as myself, it can be difficult to find the right program for each step of the bioinformatics process, and to know the correct data format and parameters used with each individual program.</p>

<p>Here is a short recap of the Pool-seq bioinformatic programs that I have gathered (in no particular order), including input and output files, and any additional notes I had on the program.</p>

<p>~</p>

<p><a href="https://sourceforge.net/p/popoolation/wiki/Manual/" target="_blank">Popoolation</a> -</p>

<p>input: a samtools pileup file (Note: must be made by samtools version 0.1.18 and below, and cannot take mpileup files)</p>

<p>output: a pool-specific sync file, from which population specific diversity indices (such at Tajima&rsquo;s pi &amp; D, Watterson&rsquo;s theta) can be calculated with the Variance-sliding.pl</p>

<p><a href="https://sourceforge.net/p/popoolation2/wiki/Manual/" target="_blank">Popoolation2</a> -</p>

<p>input: a samtools mpileup file (Note: must be made by samtools version 0.1.18 and below)</p>

<p>output: a sync file including multiple pools, from which diversity metrics (FSTs) can be calculated. The sync file can also be converted into a genepop file for further analysis.</p>

<p>Notes of Popoolation/Polpoolation2:</p>

<p>The sync2GenePop conversion step requires custom scripts to be written so that it can be run on all loci at once, instead of each locus individually. Feel free to contact me if you wish to see perl scripts used to create a GenePop file for an entire pool.</p>

<p><a href="https://cran.r-project.org/web/packages/poolfstat/poolfstat.pdf" target="_blank">poolfstat</a> -</p>

<p>input: a sync file from Popoolation2, VCF file, Baypass or Selestim input files</p>

<p>output: global and per-SNP FST values, a pairwise FST matrix, Baypass or Selestim input files</p>

<p>Notes: Remember that the poolsize list is haploid!</p>

<p><a href="http://www1.montpellier.inra.fr/CBGP/software/baypass/" target="_blank">BayPass</a> -</p>

<p>input: 1. pooled read count data, 2. poolsize file, 3. population name file (all of which can be output by poolfstat R package)</p>

<p>output: This program is used to find either FST-based or environmental-based outlier loci.</p>

<p>If you want FST-based outliers, you would run the core model. Then run the R code under section 5.1.1 of the manual. Then you will generate a POD genofile with the &lsquo;simulate.baypass&rsquo; R script, which you will use as the input in another core model. Then you will compare the outputs of the 2 core model runs with the XTX calibration R scripts.</p>

<p>If you want environmental-based outliers, then you would run the auxiliary model. This will give you an &lsquo;summary_betai&rsquo; output file. Within this file, if the BF(dB) column has a value &gt; 20, then you that locus can be identified as an outlier, corresponding to the given environmental predictor in the first column.</p>

<p>If you need any help with BayPass, I am happy to share my scripts upon request.</p>

<p><a href="https://github.com/ekg/freebayes" target="_blank">FreeBayes</a> -</p>

<p>input: BAM files &amp; a reference fasta file</p>

<p>output: a pool VCF file</p>

<p><a href="https://github.com/vibansal/crisp" target="_blank">CRISP</a> -</p>

<p>input: BAM files, pool sample sizes, reference fasta file (NB: requires at least 2 pools/2 BAM files)</p>

<p>output: a pool VCF file</p>

<p><a href="http://snver.sourceforge.net/manual.html" target="_blank">SNVer</a> -</p>

<p>input: BAM files &amp; a text file with the BAM file names, number of haploids, number of samples, and filtering thresholds</p>

<p>output: a pool VCF file</p>

<p>Notes on FreeBayes, CRISP &amp; SNVer:</p>

<p>Even though these programs make VCF files, I personally could not find a way to convert these VCF files to other formats such as genepop/genind/genlight with conversion tools such as PGDspider, vcf2genepop, or genomic_converter. However, it seems like it can be input into the R package <a href="https://cran.r-project.org/web/packages/poolfstat/poolfstat.pdf" target="_blank">poolfstat</a> to calculate F-statistics, and be converted into files for BayPass and SelEstim.</p>

<p><a href="https://github.com/EmanueleRaineri/snape-pooled" target="_blank">SNAPE-pool</a> -</p>

<p>input: a samtools pileup file (not mpileup)</p>

<p>output: allele frequency output files</p>

<p><a href="https://github.com/lucaferretti/npstat" target="_blank">npstat</a> -</p>

<p>input: a samtools mpileup file + individual BAM files (per pool). Can also take GFF3 annotations to perform McDonald-Krietman tests.</p>

<p>output: population specific diversity indices (such at Tajima&rsquo;s pi &amp; D, Watterson&rsquo;s theta), and population differentiation (FSTs)</p>

<p><a href="https://github.com/chasewnelson/SNPGenie" target="_blank">SNPgenie</a> -</p>

<p>input: a VCF file (for a single pool), a reference fasta file, and GTF annotation file</p>

<p>output: πN/πS, dN/dS, and other evolutionary parameters (has a great number output files)</p>

<p>Note: This program is only for model species with annotated genomes.</p>

<p><a href="https://github.com/bcm-uga/pcadapt" target="_blank">PCAdapt</a> -</p>

<p>input:  a frequency matrix with n rows and L columns (where n is the number of populations and L is the number of genetic markers)</p>

<p>output: several plots and list of potential outlier loci</p>

<p>Note: For my data, this program suggested quite a large number of putative outliers, especially compared to Bayescan, Loistan, and Popoolation.</p>

<p><a href="http://www1.montpellier.inra.fr/CBGP/software/selestim/files/SelEstim_1.1.7_manual.pdf" target="_blank">SelEstim</a> -</p>

<p>input: an allele/read count file</p>

<p>ouput: a &lsquo;delta&rsquo; output file which is input in R to identify outlier loci</p>

<p>Note: I personally tried SelEstim with ~ 8,000 loci, and the program ran for about 4 weeks without finishing, so could not use due to time restraints.</p>

<p><a href="https://github.com/magicDGS/PoolHMM/blob/master/README.txt" target="_blank">Pool-hmm</a> -</p>

<p>input: a samtools mpileup (Note: the mpileup command should not have -u or -g option)</p>

<p>output: allele frequency spectrum, plus probability of &ldquo;Neutral&rdquo;, &ldquo;Intermediate&rdquo; and &ldquo;Selection&rdquo; for each polymorphic site, with additional stats files</p>

<p>~</p>

<p>Additionally, here is a short list of some pipelines available for Pool-seq &amp; non-model organisms:</p>

<p><a href="https://github.com/StevenMicheletti/poolparty" target="_blank">Pool-party</a></p>

<p><a href="https://github.com/jpuritz/dDocent" target="_blank">dDocent</a></p>

<p><a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.php" target="_blank">GATK</a></p>

<p><a href="http://catchenlab.life.illinois.edu/stacks/" target="_blank">STACKS</a></p>

<p>Also, for complete bioinformatics beginners, <a href="https://usegalaxy.org/" target="_blank">Galaxy</a> is a great way to understand various bioinformatic programs and their command lines.</p>

<p>~</p>

<p>Happy analysing!</p>

    </div>

    





    
    

    

    
<section id="comments">
  <div id="disqus_thread"></div>
<script>
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "esnielsen-github-io-1" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</section>



  </div>
</article>

<footer class="site-footer">
  <div class="container">

    
    <p class="powered-by">
      <a href="https://esnielsen.github.io/privacy/">Privacy Policy</a>
    </p>
    

    <p class="powered-by">

      &copy; 2018 &middot; 

      Powered by the
      <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
      <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <button type="button" class="close btn-large" data-dismiss="modal">&times;</button>
        <h4 class="modal-title">Cite</h4>
      </div>
      <div>
        <pre><code class="modal-body tex"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-primary btn-outline js-copy-cite" href="#" target="_blank">
          <i class="fa fa-copy"></i> Copy
        </a>
        <a class="btn btn-primary btn-outline js-download-cite" href="#" target="_blank">
          <i class="fa fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha512-3P8rXCuGJdNZOnUx/03c1jOTnMn3rP63nBip5gOP2qmUh5YAdVAvFZ1E+QLZZbC1rtMrQb+mah3AfYW11RUrWA==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
        
      

      
      
    

    <script src="/js/hugo-academic.js"></script>
    

    
    

    
    
    
    <script id="dsq-count-scr" src="//esnielsen-github-io-1.disqus.com/count.js" async></script>
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    <script src="/js/search.js"></script>
    

    
    

  </body>
</html>

